docs/src/main/asciidoc/native.adoc

```diff
@@ -0,0 +1,107 @@
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#google-cloud-key-management-service[Cloud KMS]
+
+=== Usage
+
+This section provides a quick guide to build native image for Spring Boot Application with Google Cloud integrations.
+For more details, please refer to Spring boot guide to https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#native-image.developing-your-first-application[Developing Your First GraalVM Native Application]
+
+==== Paketo Buildpacks
+
+Doesn't require GraalVM to be installed on the local environment.
+Needs docker to be installed and the following
+
+1. spring-boot-starter-parent
+
+[source]
+----
+<parent>
+<groupId>org.springframework.boot</groupId>
+<artifactId>spring-boot-starter-parent</artifactId>
+</parent>
+----
+
+2. native-maven-plugin
+
+[source]
+----
+<plugin>
+<groupId>org.graalvm.buildtools</groupId>
+<artifactId>native-maven-plugin</artifactId>
+</plugin>
+----
+
+3. Call `mvn -Pnative spring-boot:build-image`
+
+==== Native Build Plugins
+
+Needs GraalVM to be installed on the local environment and the following:
+
+1. `org.springframework.boot:spring-boot-starter-parent`.
+No special dependencies for native image support.
+It is all baked into the spring boot dependency.
+This dependency provides the `native` profile.
+2. The `spring-boot-maven-plugin` - provides Spring Boot support in Maven and inherits a lot of default configurations from the spring-boot-starter-parent.
+
+[source]
+----
+<plugin>
+<groupId>org.springframework.boot</groupId>
+<artifactId>spring-boot-maven-plugin</artifactId>
+</plugin>
+----
+
+3. `org.graalvm.buildtools: native-maven-plugin`.
+The spring boot maven plugin uses this to activate native image compilation.
+4. Call `mvn -Pnative native:compile`
+
+=== Guides Specific to a Module [[detail_guides]]
+
+==== Datastore
+
+To use `DatastoreTemplate` methods on Domain types not managed by `DatastoreRepository`, you need to register runtime hints for these domain POJOs in your application. For more details, refer to example
+https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/main/java/com/example/DomainTypeRuntimeHints.java[`DomainTypeRuntimeHints`] in Datastore basic sample.
+
+===== Relationships
+@LazyReference is not yet supported for Native Image.
+
+==== Firestore
+
+To use `FirestoreTemplate`  methods on Domain types not managed by `FirestoreReactiveRepository`, you need to register runtime hints for these domain POJOs in your application. For more details, refer to example https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/java/com/example/DomainTypeRuntimeHints.java[`DomainTypeRuntimeHints`] in Reactive Firestore Repository sample application.
```

Comment:
Similarly, I thought these were `@Document` classes? Is "Domain types" clear enough for customers to understand the set of classes that will need hints?